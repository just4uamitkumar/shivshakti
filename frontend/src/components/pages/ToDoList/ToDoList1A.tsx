import {
  Box,
  Checkbox,
  Divider,
  Grid,
  List,
  ListItem,
  ListItemText,
  Stack,
  TextField,
} from "@mui/material";
import { useState } from "react";
import TypoGraphy from "../../common/Typography";
import CustomBtn from "../../common/Button";

const ToDoList1A: React.FC = () => {
  const [inputItem, setInputItem] = useState<string>("");
  const [todoList, setTodoList] = useState<string[]>([]);
  const [checkedItems, setCheckedItems] = useState<{ [key: string]: boolean }>(
    {}
  );

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setCheckedItems({
      ...checkedItems,
      [event.target.name]: event.target.checked,
    });
  };

   const handleDelete = (item:string) => {
    const newArr = todoList.filter((todo) => todo !== item);
    setTodoList(newArr);
  };

  return (
    <>
      <Grid spacing={2} className="p-3">
        <Box className="p-1">
          <TypoGraphy variant="h2" component={"h2"}>
            {"To Do List 1 A"}
          </TypoGraphy>
        </Box>
        <Divider />

        <Box className="p-3">
          <Grid size={12}>
            <Stack flexDirection="row" gap={2} alignItems="center">
              <TextField
                label="Add Item"
                variant="outlined"
                value={inputItem}
                onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                  setInputItem(e.target.value)
                }
              />
              <CustomBtn
                text="Add Value"
                variant="contained"
                className="primary-btn"
                onClick={() => {
                  setTodoList([...todoList, inputItem]);
                  setInputItem("");
                }}
              />
            </Stack>
          </Grid>
          <Grid size={12} className={"pb-3"}>
            <List>
              {todoList.map((item, index) => (
                <ListItem key={index}>
                  <Checkbox
                    name={item}
                    checked={checkedItems[item] || false}
                    onChange={handleChange}
                  />
                  <ListItemText primary={item} />

                  {checkedItems[item] && (
                    <CustomBtn
                      text="Delete"
                      variant="contained"
                      className="cancel-btn sm-btn"
                      onClick={() => handleDelete(item)}
                    />
                  )}
                </ListItem>
              ))}
            </List>
          </Grid>
        </Box>
      </Grid>
    </>
  );
};

export default ToDoList1A;
