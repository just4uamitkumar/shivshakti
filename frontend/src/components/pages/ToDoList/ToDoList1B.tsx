import { Box, Divider, Grid } from "@mui/material";
import TypoGraphy from "../../common/Typography";

const ToDoList1B: React.FC = () => {
  return (
    <>
      <Grid size={12}>
        <Box className="p-1">
          <TypoGraphy variant="h2" component={"h2"}>
            {"Number Problem 2"}
          </TypoGraphy>
        </Box>
        <Divider />

        <Box className="p-3"></Box>
      </Grid>
    </>
  );
};

export default ToDoList1B;
