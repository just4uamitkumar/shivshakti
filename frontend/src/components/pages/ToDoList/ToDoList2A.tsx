import { Box, Divider, Stack, Grid } from "@mui/material";
import TypoGraphy from "../../common/Typography";
import { useEffect, useState } from "react";

const ToDoList2A: React.FC = () => {
  const [string1, setString1] = useState<string>("");
  const [isParaDim, setIsParaDim] = useState<boolean>(false);
  const [string2, setString2] = useState<object>({
    firstStr: "",
    secondStr: "",
  });
  const [string3, setString3] = useState<object>({});
  const [string4, setString4] = useState<string>("");

  useEffect(() => {
    const mystr: string = "delhi is capital of india";

    setString1(reverseString(mystr));

    const paraDom: string = "amit";
    setIsParaDim(paraDomString(paraDom));

    const word: string = "this is india";
    setString2({
      firstStr: capitalLett(word).myStr1,
      secondStr: capitalLett(word).myStr2,
    });

    const str: string = "apppplleenccd";
    setString3(checkStringOccurence(str));
  }, []);

  useEffect(() => {
    setString4(mostFrequentValue(string3));
  }, [string3]);

  //Reverse a string
  const reverseString = (str: string) => {
    const myStr = str
      ?.split(" ")
      .map((item: string) => item.split("").reverse().join(""))
      .join(" ");
    return myStr;
  };

  //Capitilize first letter of a word
  const capitalLett = (str: string) => {
    const myStr1: string =
      str.split("")[0].charAt(0).toUpperCase() + str.substring(1);
    const myStr2: string = str
      .split(" ")
      .map((elem: string) => elem.charAt(0).toUpperCase() + elem.substring(1))
      .join(" ");
    return {
      myStr1,
      myStr2,
    };
  };

  //Paradimonal a string
  const paraDomString = (str: string) => {
    return str === str.split("").reverse().join("");
  };

  //check string occurence
  const checkStringOccurence = (str: string) => {
    const obj:unknown = {};
    str.split("").forEach((elem: string) => {
      obj.hasOwnProperty(elem) ? obj[elem]++ : (obj[elem] = 1);
    });
    return obj;
  };

  //Most frequest value in Array
  const mostFrequentValue = (myObj: object) => {
    const keyArr: string[] = Object.keys(myObj);
    const valueArr: string[] = Object.values(myObj);
    const newValArr: string[] = valueArr.map((e) => e);
    const hightstValue = newValArr.sort((a:string, b:string) => a - b)[valueArr.length - 1];
    const valueIndex:number = valueArr.indexOf(hightstValue);
    return `${keyArr[valueIndex]}  found ${hightstValue} times`;
  };
  return (
    <>
      <Grid size={12}>
        <Box className="p-1">
          <TypoGraphy variant="h2" component={"h2"}>
            {"String Problem 1"}
          </TypoGraphy>
        </Box>
        <Divider />

        <Box className="p-3">
          <Grid size={12} className={"pb-3"}>
            <TypoGraphy variant="h4" component={"h4"} typeClass={"py-1"}>
              {"Show Reverse String"}
            </TypoGraphy>
            <Stack>{string1}</Stack>
          </Grid>
          <Grid size={12} className={"pb-3"}>
            <TypoGraphy variant="h4" component={"h4"} typeClass={"py-1"}>
              {"Paridominal String"}
            </TypoGraphy>
            <Stack>{isParaDim ? "true" : "false"}</Stack>
          </Grid>
          <Grid size={12} className={"pb-3"}>
            <TypoGraphy variant="h4" component={"h4"} typeClass={"py-1"}>
              {"Capitilize first letter"}
            </TypoGraphy>
            <Stack>{string2.firstStr}</Stack>
            <Stack>{string2.secondStr}</Stack>
          </Grid>
          <Grid size={12} className={"pb-3"}>
            <TypoGraphy variant="h4" component={"h4"} typeClass={"py-1"}>
              {"Check String Occurence"}
            </TypoGraphy>
            <Stack>{Object.entries(string3)}</Stack>
            <Stack>{string4}</Stack>
          </Grid>
        </Box>
      </Grid>
    </>
  );
};

export default ToDoList2A;
