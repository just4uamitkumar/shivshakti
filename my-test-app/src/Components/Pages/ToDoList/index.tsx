import { useState } from "react";

const ToDoList = () => {
  const [todos, setTodos] = useState<string[]>([]);
  const [input, setInput] = useState<string>("");

  const addTodo = () => {
    if (input.trim() !== "") {
      setTodos([...todos, input]);
      setInput("");
    }
  };

  const deleteToDo = (index: number) => {
    setTodos(todos.filter((_, i) => i !== index))
  }

  return (
    <div className="todo-app">
      <h1>{"To Do List"}</h1>
      <div className="input-container">
        <input
          data-testId="todo-input"
          placeholder="Enter a new todo"
          value={input}
          role="textbox"
          onChange={(e) => setInput(e.target.value)}
        />
         <button data-testId="add-button" onClick={addTodo}>Add Todo </button>
      </div>
      <ul className="todo-list">
        {todos.map((todo, index) => (
          <li key={index} className="todo-item" data-testId={`todo-item-${index}`}>
            {todo}
            <button
              data-testId={`remove-button-${index}`}
              className="remove-button"
              onClick={() => deleteToDo(index)} >{'Delete'}</button>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default ToDoList;
