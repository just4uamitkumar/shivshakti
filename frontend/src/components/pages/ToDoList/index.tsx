import PageBanner from "../../shared/PageBanner";
import { useState } from "react";
import { Box, Grid } from "@mui/material";
import TypoGraphy from "../../common/Typography";
import ToDoList1A from "./ToDoList1A";
import ToDoList1B from "./ToDoList1B";
import ToDoList2A from "./ToDoList2A";
import ToDoList2B from "./ToDoList2B";

type page = {
  name: string;
  comp: React.FC;
};

const ToDoList: React.FC = () => {
  const pageObj: page[] = [
    { name: "ToDoList1A", comp: () => <ToDoList1A /> },
    { name: "ToDoList1B", comp: () => <ToDoList1B /> },
    { name: "ToDoList2A", comp: () => <ToDoList2A /> },
    { name: "ToDoList2A", comp: () => <ToDoList2B /> },
  ];

  const [currentPage, setCurrentPage] = useState<React.FC>(pageObj[0].comp);

  const getPage = (item: string) => {
    const page = pageObj.find((e) => e.name === item);
    setCurrentPage(page?.comp ?? pageObj[0].comp);
  };

  return (
    <>
      <PageBanner title={"To Do List"} />
      <Grid className={"contentWraper"}>
        <Grid className="container" container>
          <Grid className={"leftSideBar"}>
            <Box className="pl-2 pb-2">
              <TypoGraphy variant={"h6"}>{"To Do List 1"}</TypoGraphy>
              <ul className="left-links pl-1">
                <li onClick={() => getPage(pageObj[0].name)}>
                  To Do List 1 A
                </li>
                <li onClick={() => getPage(pageObj[1].name)}>To Do List 1 B</li>
              </ul>
            </Box>
            <Box className="pl-2 pb-2">
              <TypoGraphy variant={"h6"}>{"To Do List 2"}</TypoGraphy>
              <ul className="left-links pl-1">
                <li onClick={() => getPage(pageObj[2].name)}>
                  {"To Do List 2 A"}
                </li>
                <li onClick={() => getPage(pageObj[3].name)}>
                  {" "}
                  {"To Do List 2 B"}
                </li>
              </ul>
            </Box>
          </Grid>
          <Grid className="rightSection">{currentPage}</Grid>
        </Grid>
      </Grid>
    </>
  );
};

export default ToDoList;
