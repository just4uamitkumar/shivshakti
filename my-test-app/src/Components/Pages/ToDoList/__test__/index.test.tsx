import { afterEach, beforeEach, describe, expect, it } from "vitest";
import ToDoList from "../index";
import { fireEvent, render, screen } from "@testing-library/react";

describe("ToDoList Component", () => {
  beforeEach(() => {});

  afterEach(() => {});

  it("matches snapshot", () => {
    const { container } = render(<ToDoList />);
    expect(container).toMatchSnapshot();
  });

  it("should render the component", () => {
    expect(ToDoList).toBeDefined();
  });

  it("should have a headline", () => {
    render(<ToDoList />);
    const heading = screen.getByText("To Do List");
    expect(heading).toBeInTheDocument();
  });

  it("should have an input field", () => {
    render(<ToDoList />);
    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  it("should have an input field 2 method", () => {
    render(<ToDoList />);
    const inputField = screen.getByTestId("todo-input");
    expect(inputField).toBeInTheDocument();
  });

  it("should have an add Button", () => {
    render(<ToDoList />);
    const addButton = screen.getByTestId("add-button");
    expect(addButton).toBeInTheDocument();
  });

  it("should add the item from input box", () => {
    render(<ToDoList />);
    const inputField = screen.getByTestId("todo-input");
    fireEvent.change(inputField, { target: { value: "Buy Shoes" } });
    const addButton = screen.getByTestId("add-button");
    fireEvent.click(addButton);
    const todoItem = screen.getByText("Buy Shoes");
    expect(todoItem).toBeInTheDocument();
  });

  it("should delete the item from the list", () => {
    render(<ToDoList />);
    const inputField = screen.getByTestId("todo-input");
    fireEvent.change(inputField, { target: { value: "Buy Shoes" } });
    const addButton = screen.getByTestId("add-button");
    fireEvent.click(addButton);

    const deleteButton = screen.getByTestId("remove-button-0");
    fireEvent.click(deleteButton);

    expect(screen.queryByText("Buy Shoes")).not.toBeInTheDocument();
  });

  // Additional tests can be added here to check functionality, state changes, etc.
});
