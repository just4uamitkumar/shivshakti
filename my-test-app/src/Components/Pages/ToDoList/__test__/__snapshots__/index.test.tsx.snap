// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ToDoList Component > matches snapshot 1`] = `
<div>
  <div
    class="todo-app"
  >
    <h1>
      To Do List
    </h1>
    <div
      class="input-container"
    >
      <input
        data-testid="todo-input"
        placeholder="Enter a new todo"
        role="textbox"
        value=""
      />
      <button
        data-testid="add-button"
      >
        Add Todo 
      </button>
    </div>
    <ul
      class="todo-list"
    />
  </div>
</div>
`;
